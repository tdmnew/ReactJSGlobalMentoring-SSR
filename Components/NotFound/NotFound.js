import React from 'react';
import { useTranslation } from 'react-i18next';
import Link from 'next/link';

import { TRANSLATIONS } from '../../Core/I18N';

export default function NotFound() {
  const title = TRANSLATIONS.EN.APP_NAME.split(' ');

  return (
    <div className="not-found">
      <div className="not-found heading">
        <p className="heading logo">
          <span className="heading logo--left">{title[0]}</span>
          <span className="heading logo--right">{title[1]}</span>
        </p>
      </div>
      <div className="not-found body">
        <h3 className="not-found body__text--top">
          {TRANSLATIONS.EN.PAGE_NOT_FOUND}
        </h3>
        <h4 className="not-found body__text--bottom">
          {TRANSLATIONS.EN.FOUR_OH_FOUR}
        </h4>
        <Link className="not-found body__btn" href="/">
          {TRANSLATIONS.EN.GO_HOME}
        </Link>
      </div>
    </div>
  );
}
