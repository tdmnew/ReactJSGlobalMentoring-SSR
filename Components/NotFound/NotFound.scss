.not-found {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;
    color: white;

    .heading {
        display: flex;
        align-items: center;
        height: 5vh;
        width: 100%;
        .logo {
            align-self: flex-start;
            font-size: 20px;
            color: #f65261;
            width: auto;
            padding-left: 2rem;

            &--left {
                font-weight: bold;
            }
            &--right {
                font-weight: lighter;
            }
        }
    }

    .body {
        display: flex;
        align-items: center;
        height: 90vh;
        width: 100%;
        &__text {
            h4 {
                -moz-text-fill-color: #fff;
                -webkit-text-fill-color: #fff;
                -moz-text-stroke-color: #f65261;
                -webkit-text-stroke-color: #f65261;
                -moz-text-stroke-width: 3px;
                -webkit-text-stroke-width: 3px;
            }
            &--top {
                position: relative;
                left: 5px;
                width: 20rem;
                font-size: 1.5rem;
                font-weight: lighter;
                letter-spacing: 5px;
            }

            &--bottom {
                width: 20rem;
                text-align: center;
                margin: 1rem 0;
                display: block;
                font-size: 10rem;
                animation: glitch 1s infinite;
            }
        }

        &__btn {
            height: 2rem;
            width: 10rem;
            font-weight: lighter;
            text-decoration: none;
            line-height: 27px;
            border-radius: 3px;
            color: #f65261;
            border: 1px solid #f65261;
            background: none;
            cursor: pointer;
            position: relative;
            padding: 0 1rem;
            margin: 1rem 0;
        }
    }

    @keyframes glitch {
        2%,
        64% {
            transform: translate(2px, 0) skew(0deg);
        }
        4%,
        60% {
            transform: translate(-2px, 0) skew(0deg);
        }
        62% {
            transform: translate(0, 0) skew(5deg);
        }
    }
}
