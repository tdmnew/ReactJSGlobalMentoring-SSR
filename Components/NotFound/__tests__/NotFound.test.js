import React from "react";

import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import NotFound from "../NotFound.js";
import store from "../../../Redux/index.js";

it("Renders the not found page matching snapshot", () => {
    const tree = render(
        <Provider store={store}>
            <MemoryRouter>
                <NotFound />
            </MemoryRouter>
        </Provider>
    );
    expect(tree).toMatchSnapshot();
});
