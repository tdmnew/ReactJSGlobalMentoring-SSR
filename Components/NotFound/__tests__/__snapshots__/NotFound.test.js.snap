// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the not found page matching snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="not-found"
      >
        <div
          class="not-found heading"
        >
          <p
            class="heading logo"
          >
            <span
              class="heading logo--left"
            >
              netflix
            </span>
            <span
              class="heading logo--right"
            >
              roulette
            </span>
          </p>
        </div>
        <div
          class="not-found body"
        >
          <h3
            class="not-found body__text--top"
          >
            Page Not Found
          </h3>
          <h4
            class="not-found body__text--bottom"
          >
            404
          </h4>
          <a
            aria-current="page"
            class="not-found body__btn active"
            href="/"
          >
            GO BACK TO HOME
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="not-found"
    >
      <div
        class="not-found heading"
      >
        <p
          class="heading logo"
        >
          <span
            class="heading logo--left"
          >
            netflix
          </span>
          <span
            class="heading logo--right"
          >
            roulette
          </span>
        </p>
      </div>
      <div
        class="not-found body"
      >
        <h3
          class="not-found body__text--top"
        >
          Page Not Found
        </h3>
        <h4
          class="not-found body__text--bottom"
        >
          404
        </h4>
        <a
          aria-current="page"
          class="not-found body__btn active"
          href="/"
        >
          GO BACK TO HOME
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
